docs/semantics/regex.md

```diff
@@ -0,0 +1,53 @@
+# Regex Semantic
+
+## Overview
+
+A regular expression type is an expression used for pattern matching. A regular expression type is declared using `regx` identifier, always begins with `r`, and is bounded by two single quotes `'<pattern>'` where `<pattern>` is the desired pattern to be matched. `regx` type is constant once initialized and cannot be changed.
+
+There are no operations on `regx` type but there are native functions that require a `regx` type as an argument, ie `filter`
+
+## Syntax
+
+A `regx` type declaration in general takes the form of `regx myPattern = r'<pattern>'` where `<pattern>` is the desired pattern.
+
+## Semantics
+
+A `regx` matches against an entire `string`. Therefore any attempt to `filter` against an array of strings will return all strings that the `regx` matches against.
+
+### Operators

```

Comment:
Is there a way to emulate the '?' operator using only these operators? If not, maybe we can add it? i've always found that to be a very useful operator.

Reply:
It'll all be in the library

this page is mostly just emulating the gnu page